mate-panel/panel-run-dialog.c

```diff
@@ -1962,6 +2030,20 @@ panel_run_dialog_new (GdkScreen  *screen,
 	g_signal_connect_swapped (dialog->run_dialog, "destroy",
 				  G_CALLBACK (panel_run_dialog_destroy), dialog);
 
+
+        for (int i = 0; accelerator_key_mapping[i].list_item_idx != -1; i++) {
+              GClosure* closure_key = g_cclosure_new(
```

Comment:
Do I need to manually destroy the objects created by g_cclosure_new and gtk_accel_group_new?

Reply:
* `g_cclosure_new()`: No, because [`gtk_accel_group_new()`](https://gitlab.gnome.org/GNOME/gtk/-/blob/gtk-3-24/gtk/gtkaccelgroup.c#L627) [sinks](https://docs.gtk.org/gobject/method.Closure.sink.html) the `GClosure` reference.
* `gtk_accel_group_new()`: I'm less positive, but it *seems* to have to be unreffed manually: it's not a `GInitiallyUnowned`, and without additional info in the docs it should mean you have to manage your reference manually, and it seems the code indeed adds its own reference when attaching it to the window.

This said, shouldn't you create only *one* `GtkAccelGroup`, and connect all the keys on it?  After all, it's not called a *group* for nothing :wink:

Reply:
I factored the gtk_accel_group_new call outside the loop. I could not find gtk_accel_group_free in the documentation, so I guess it's not a memory leak.